pkg/occlient/templates.go

```diff
@@ -139,6 +148,35 @@ func generateSupervisordDeploymentConfig(commonObjectMeta metav1.ObjectMeta, com
 	return dc
 }
 
+func updateSupervisorDeploymentConfig(updateParams SupervisorDUpdateParams) appsv1.DeploymentConfig {
+
+	dc := *updateParams.existingDc
+
+	dc.ObjectMeta = updateParams.commonObjectMeta
+
+	if len(dc.Spec.Template.Spec.Containers) > 0 {
+		dc.Spec.Template.Spec.Containers[0].Name = updateParams.commonObjectMeta.Name
+		dc.Spec.Template.Spec.Containers[0].Ports = updateParams.commonImageMeta.Ports
+		dc.Spec.Template.Spec.Containers[0].Env = updateParams.envVar
+		dc.Spec.Template.Spec.Containers[0].EnvFrom = updateParams.envFrom
+	}
+
+	containerIndex := -1
```

Comment:
Never really seen a `-1` been used.. Mind explaining in a comment why?

Reply:
I used it as the container index cannot be -1 and since we can store the container index in the same variable. But I am removing this logic due to https://github.com/openshift/odo/pull/1684#discussion_r282535213